src/tests/exprtype/exprtype_test.go

```diff
@@ -1868,6 +1868,56 @@ exprtype($tuple_int_str, "unknown_from_list");
 	runExprTypeTest(t, &exprTypeTestParams{code: code})
 }
 
+func TestPropertyTypeHints(t *testing.T) {
+	code := `<?php
+class Boo {}
+
+
+class Foo {
+  public static int $int;
+  public static float $float;
+  public static string $string;
+  public static bool $bool;
+  public static array $array;
+  public static callable $callable;
+  public static iterable $iterable;
+  public static object $object;
```

Comment:
I would replace at least some of these test cases with non-static typed properties tests.